dataloader/dataloader_impl.hpp

```diff
@@ -145,4 +145,137 @@ template<
   }
 }
 
+template<
+  typename DatasetX,
+  typename DatasetY,
+  class ScalerType
+> void DataLoader<
+    DatasetX, DatasetY, ScalerType
+>::LoadObjectDetectionDataset(const std::string& pathToAnnotations,
+                              const std::string& pathToImages,
+                              const std::vector<std::string>& classes,
+                              const std::vector<std::string>& augmentations,
+                              const double augmentationProbability,
+                              const bool absolutePath,
+                              const std::string& baseXMLTag,
+                              const std::string& imageNameXMLTag,
+                              const std::string& sizeXMLTag,
+                              const std::string& objectXMLTag,
+                              const std::string& bndboxXMLTag,
+                              const std::string& classNameXMLTag,
+                              const std::string& x1XMLTag,
+                              const std::string& y1XMLTag,
+                              const std::string& x2XMLTag,
+                              const std::string& y2XMLTag)
+{
+  Augmentation<DatasetX> augmentation(augmentations, augmentationProbability);
+
+  std::vector<boost::filesystem::path> annotationsDirectory, imagesDirectory;
+
+  // Fill the directory.
+  Utils::ListDir(pathToAnnotations, annotationsDirectory, absolutePath);
+  Utils::ListDir(pathToImages, imagesDirectory, absolutePath);
+
+  // Create a map for labels and corresponding class name.
+  // This provides faster access to class labels.
+  std::unordered_map<std::string, size_t> classMap;
+  for (size_t i = 0; i < classes.size(); i++)
+  {
+    classMap.insert(std::make_pair(classes[i], i));
+  }
+
+  // Map to insert values in a column vector.
+  std::unordered_map<std::string, size_t> indexMap;
+  indexMap.insert(std::make_pair(classNameXMLTag, 0));
+  indexMap.insert(std::make_pair(x1XMLTag, 1));
+  indexMap.insert(std::make_pair(y1XMLTag, 2));
+  indexMap.insert(std::make_pair(x2XMLTag, 3));
+  indexMap.insert(std::make_pair(y2XMLTag, 4));
+
+  // Read the xml file.
+  for (boost::filesystem::path annotationFile : annotationsDirectory)
+  {
+    if (annotationFile.string().length() <= 3 ||
+        annotationFile.string().substr(
+            annotationFile.string().length() - 3) != "xml")
+    {
+      continue;
+    }
+    // Read the xml file.
+    boost::property_tree::ptree xmlFile;
+    boost::property_tree::read_xml(annotationFile.string(), xmlFile);
+
+    // Get annotation from xml file.
+    boost::property_tree::ptree annotation = xmlFile.get_child(baseXMLTag);
+
+    // Read properties inside annotation file.
+    // Get image name.
+    std::string imgName = annotation.get_child(imageNameXMLTag).data();
+
+    // If image doesn't exist then skip the current xml file.
```

Comment:
```suggestion
    // If image doesn't exist then skip the current XML file.
```

Reply:
Sorry about the style changes, Will try to be more consistent with wording. Thanks a lot for the review. Making the changes now.